### PR TITLE
Class intf refactor - Issue #21

### PR DESCRIPTION
**Pull request by EduardoCopat**
Hello,

This is an initial version of the CLAS into INTF refactoring (https://github.com/larshp/abapGit/issues/21).

I started just with the _deserialization_ method. In order to refactor, I needed tests, in order to have tests, I had to decouple several dependencies. It was a lot of work. Hence why I just started with only one method.

Calls to function modules where isolated behind an _interface_, in this way I could provide a factory to this functions and create test doubles.

I'd like some feedback whether I'm going into the right direction.

### CODE REVIEW

File: src/zabapgit_object_clas.prog.abap [hunk 1]
```diff
@@ -875,202 +1170,671 @@ CLASS lcl_object_clas IMPLEMENTATION.
...
+
+  METHOD lif_object_oriented_object~create_documentation.
+    mv_docu_object_name = iv_object_name.
+    mv_docu_language    = iv_language.
+    mt_docu_lines       = it_lines.
+  ENDMETHOD.
 
+ENDCLASS.
+
+CLASS ltd_fake_object_files DEFINITION FOR TESTING
```
Comment (larshp) [inline src/zabapgit_object_clas.prog.abap]: please prefix the test classes with `ltcl_`, and can we move the tests to a separate include?

**Comment by larshp** [comment]
it looks very nice, perhaps too nice if that is possible 😄 

I'm all for dependency injection etc, but it will make some of the other code look bad 😁

File: src/zabapgit_object_clas.prog.abap (hunk 1 above)
Comment (EduardoCopat) [inline src/zabapgit_object_clas.prog.abap]: Yes, that can be done. Should I move everything to the unit test include or rather create a new test include?

**Comment by EduardoCopat** [comment]
@larshp Hahaha, abapGit code is not bad, especially when you compare to the average ABAP code...

**Comment by larshp** [comment]
Some of it is bad, but I think it is more important to just get the code out there. If there are important parts like CLAS/INTF, we can always refactor as you are doing. Shipping abapGit can be done very quickly so if there are any problems they can easily be fixed

**Comment by EduardoCopat** [comment]
@larshp Code was adjusted :)

**Comment by larshp** [comment]
yep, LGTM, thanks

**Pull request merged**